lib/Test/Nginx/Util.pm

```diff
@@ -578,6 +604,19 @@ sub cleanup () {
         kill_process($ChildPid, 1);
         undef $ChildPid;
     }
+
+    if ($Randomize) {
+        if (-d $ServRoot && $ServRoot =~ m{/t/servroot_\d+}) {
+            if (-f $PidFile) {
+                my $pid = get_pid_from_pidfile('');
+
+                if (defined $pid && $pid ne '' && is_running($pid)) {
+                    kill_nginx($pid, '');
```

Comment:
This is duplicate since `END {}` already cleans up nginx processes. The problem here is that `clean()` gets called before the nginx process cleanup code in `END {}` so at that point in `END {}`, the pid file no longer exists. I've just fixed this in my latest `randomize` branch. Please have a look. Thanks!